### DISCUSSION TITLE
How to get most recent changes in a conda installation?

### DISCUSSION THREAD

**Discussion by williamhobbs** [blaylockbk/Herbie#250] [Q&A] [answered]
I haven't been able to get all of the dependencies for Herbie working with `pip`, so I've used Conda/miniforge, which has been great. I may soon want to help a colleague get setup to use Herbie, but with the most recent changes (e.g., https://github.com/blaylockbk/Herbie/pull/248). 

Is there a simple way to get the recent changes in a Conda installation? It doesn't seem like there is a Conda equivalent to `pip install git+https://github.com/blaylockbk/Herbie.git`. Running that command inside a Conda env after installing Herbie didn't work for me, returning an error that ended with this (which I think is similar to what happened when I tried to install everything with `pip` instead of Conda originally):

```
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pygrib
  Building wheel for eccodes (setup.py) ... done
  Created wheel for eccodes: filename=eccodes-1.6.1-py3-none-any.whl size=41052 sha256=2be20cebe9debfbef70d2cf45e953650b248d21cc26fa6eacc12a2696a5f4adc
  Stored in directory: c:\users\whobbs\appdata\local\pip\cache\wheels\53\b1\c4\c18dbe22e94d42abfc93519ae60691eed4c8af9fb60fe6713f
  Building wheel for findlibs (setup.py) ... done
  Created wheel for findlibs: filename=findlibs-0.0.5-py3-none-any.whl size=6954 sha256=c5fbe5111268ce1106f28b9b6323668ecb62eb4e6a3fd717cf586aa3e5c23503
  Stored in directory: c:\users\whobbs\appdata\local\pip\cache\wheels\45\9b\ec\b90c9bd8c078a0ff3c7e6ec60c8ae54483402feb8631a38ae6
Successfully built herbie-data eccodes findlibs
Failed to build cartopy pygrib
ERROR: Could not build wheels for cartopy, pygrib, which is required to install pyproject.toml-based projects
``` 

I have very limited experience with git, but have also used `curl`. My default would be just to have my colleague manually download the relevant file(s) and replace them using Windows File Explorer, but I thought I'd check in case there is something else I'm missing. My colleague has less experience with python/Conda/git/pip/etc. than me, so I'm trying to keep it simple and easy to document and follow. 

Thanks!

**Comment by blaylockbk** [comment, marked as answer]
(I've been meaning to release a new version of Herbie to PyPi; But I'm waiting for the next release of MetPy)

> ERROR: Could not build wheels for cartopy, pygrib

What version of Python are you using? I've pretty much moved on to python 3.11 and haven't tested anything earlier.

---

I usually create conda envrionments from a file like https://github.com/blaylockbk/Herbie/blob/main/environment.yml

Instead of installing herbie-data from conda forge, you can install Herbie from GitHub (note the deletion and insertion below...)

```diff
name: herbie
channels:
  - conda-forge
dependencies:
  - python>=3.11
  - pip
  - rclone # Not used by Herbie, useful if you are downloading data (https://rclone.org/)

-  - herbie-data

  #==============
  # Dependencies
  #==============
  - curl
  - cartopy>=0.22
  - cfgrib>=0.9.10.4
  - dask
  - eccodes
  - geos
  - matplotlib>=3.5.3
  - metpy
  - netcdf4
  - numpy>=1.25
  - pandas>=2.0
  - pygrib>=2.1.4
  - pylint
  - pyproj>=3.6
  - requests>=2.31
  - s3fs
  - toml
  #- wgrib2 # ONLY AVAILABLE ON LINUX. Uncomment if you want this optional dependency.
  - jupyter
  - jupyterlab
  - xarray>=2023.7
  - zarr

  - pip:
      # Borrow some tools from my garage
      - git+https://github.com/blaylockbk/Carpenter_Workshop.git
+     - git+https://github.com/blaylockbk/Herbie.git
```

Then create the environment with
```
conda env create -f environment.yml
```

**Reply by williamhobbs** [reply]
Thanks. I'll give this a try. 

I'm using Python 3.12.0 on one PC (where that error came from) and 3.10.13 on another PC (where I spent the most time trying to get pip to work without Conda a while ago.

**Reply by blaylockbk** [reply]
I just published 2023.12.1 to PyPI and conda-forge. Maybe that will help you out.

**Reply by williamhobbs** [reply]
It does. Thank you!
